service/src/java/org/apache/hive/service/cli/session/SessionManager.java

```diff
@@ -135,6 +138,13 @@ public synchronized void init(HiveConf hiveConf) {
     userIpAddressLimit = hiveConf.getIntVar(ConfVars.HIVE_SERVER2_LIMIT_CONNECTIONS_PER_USER_IPADDRESS);
     LOG.info("Connections limit are user: {} ipaddress: {} user-ipaddress: {}", userLimit, ipAddressLimit,
       userIpAddressLimit);
+
+    int cleanupThreadCount = hiveConf.getIntVar(ConfVars.HIVE_ASYNC_CLEANUP_SERVICE_THREAD_COUNT);
+    int cleanupQueueSize = hiveConf.getIntVar(ConfVars.HIVE_ASYNC_CLEANUP_SERVICE_QUEUE_SIZE);
+    if (cleanupThreadCount > 0) {
+      cleanupService = new EventualCleanupService(cleanupThreadCount, cleanupQueueSize);
+    }
+    cleanupService.start();
```

Comment:
it seems to me that `cleanupService` is only initialized in case `cleanupThreadCount>0` - wouldn't there be an `NPE` when that is not satisfied?

Reply:
fixed